lib/composite.ex

```diff
@@ -387,6 +406,21 @@ defmodule Composite do
     {query, MapSet.union(loaded_deps, deps_to_load)}
   end
 
+  defp maybe_raise_on_unknown_params(%__MODULE__{strict: true} = composite) do
+    diff =
+      MapSet.difference(
+        MapSet.new(Map.keys(composite.params)),
+        MapSet.new(composite.param_definitions |> Enum.map(&elem(&1, 0)) |> List.flatten())
+      )
+
+    case MapSet.size(diff) do
+      0 -> composite
+      _ -> raise ArgumentError, "Unknown params: #{inspect(MapSet.to_list(diff))}"
+    end
+  end
+
+  defp maybe_raise_on_unknown_params(composite), do: composite
```

Comment:
`maybe_raise_on_unknown_params` function cannot modify `composite` struct, so let's return `:noop` when strict option is ignored

Reply:
Done!